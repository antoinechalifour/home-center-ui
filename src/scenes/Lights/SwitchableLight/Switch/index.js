import React from 'react'
import PropTypes from 'prop-types'
import ToggleSwitch from 'ui/ToggleSwitch'

export default function Switch ({ isOn, onClick }) {
  return <ToggleSwitch value={isOn} onChange={onClick} />
}

Switch.propTypes = {
  isOn: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
}
