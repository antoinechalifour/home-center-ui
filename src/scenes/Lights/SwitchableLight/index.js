import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { graphql, compose } from 'react-apollo'
import { switchLight, updateLightInformation } from 'queries/lights'
import SwitchableLight from './SwitchableLight'

export class SwitchableLightContainer extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    switchLight: PropTypes.func.isRequired,
    updateLightInformation: PropTypes.func.isRequired
  }

  _onToggleLight = () => {
    const isOn = this.props.status === 'on'

    this.props.switchLight({
      variables: {
        input: {
          id: this.props.id,
          isOn: !isOn
        }
      }
    })
  }

  _onUpdateName = name => {
    this.props.updateLightInformation({
      variables: {
        input: {
          id: this.props.id,
          name
        }
      }
    })
  }

  render () {
    return (
      <SwitchableLight
        {...this.props}
        toggleLight={this._onToggleLight}
        updateName={this._onUpdateName}
      />
    )
  }
}

const switchLightOptions = {
  name: 'switchLight',
  options: {
    refetchQueries: ['GetLights']
  }
}

const updateLightInformationOptions = {
  name: 'updateLightInformation',
  options: {
    refetchQueries: ['GetLights']
  }
}

export default compose(
  graphql(switchLight, switchLightOptions),
  graphql(updateLightInformation, updateLightInformationOptions)
)(SwitchableLightContainer)
