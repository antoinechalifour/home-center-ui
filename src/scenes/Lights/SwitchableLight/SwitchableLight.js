import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Editable from 'components/Editable'
import Switch from './Switch'

export default class SwitchableLight extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    toggleLight: PropTypes.func.isRequired,
    updateName: PropTypes.func.isRequired
  }

  render () {
    const isOn = this.props.status === 'on'

    return (
      <Container>
        <Name>
          <Editable onChange={this.props.updateName}>
            {this.props.name}
          </Editable>
        </Name>
        <Switch isOn={isOn} onClick={this.props.toggleLight} />
      </Container>
    )
  }
}

const Container = styled.div`
  display: flex;
  align-items: center;
  box-sizing: border-box;
`

const Name = styled.div`
  flex: 1;
`
