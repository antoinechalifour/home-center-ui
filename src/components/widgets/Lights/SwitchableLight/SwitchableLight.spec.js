import React from 'react'
import { shallow } from 'enzyme'
import SwitchableLight from './SwitchableLight'

describe('scenes/HomeControl/Lights/SwitchableLight', () => {
  it('Should render correctly (off)', () => {
    const props = {
      id: '3',
      name: 'Light #3',
      status: 'off',
      toggleLight: jest.fn(),
      updateName: jest.fn()
    }
    const wrapper = shallow(<SwitchableLight {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('Should render correctly (on)', () => {
    const props = {
      id: '3',
      name: 'Light #3',
      status: 'on',
      toggleLight: jest.fn(),
      updateName: jest.fn()
    }
    const wrapper = shallow(<SwitchableLight {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
