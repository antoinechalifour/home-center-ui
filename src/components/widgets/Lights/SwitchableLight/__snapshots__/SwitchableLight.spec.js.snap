// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/HomeControl/Lights/SwitchableLight Should render correctly (off) 1`] = `
<styled.div>
  <MdLightbulbOutline />
  <styled.div>
    <Editable
      onChange={[Function]}
    >
      Light #3
    </Editable>
  </styled.div>
  <Switch
    isOn={false}
    onClick={[Function]}
  />
</styled.div>
`;

exports[`scenes/HomeControl/Lights/SwitchableLight Should render correctly (on) 1`] = `
<styled.div>
  <MdLightbulbOutline />
  <styled.div>
    <Editable
      onChange={[Function]}
    >
      Light #3
    </Editable>
  </styled.div>
  <Switch
    isOn={true}
    onClick={[Function]}
  />
</styled.div>
`;
